pkg/flow/internal/controller/metrics.go

```diff
@@ -110,9 +110,11 @@ func (cc *controllerCollector) Collect(ch chan<- prometheus.Metric) {
 	componentsByHealth := make(map[string]int)
 
 	for _, component := range cc.l.Components() {
-		health := component.CurrentHealth().Health.String()
-		componentsByHealth[health]++
-		component.registry.Collect(ch)
+		if builtinComponent, ok := component.(*BuiltinComponentNode); ok {
+			health := builtinComponent.CurrentHealth().Health.String()
+			componentsByHealth[health]++
```

Comment:
The health applies to all ComponentNodes, so I think this needs to be done outside of the type check so we collect health information on non-builtin components.

Only the call to `builtinComponent.registry.Collect(ch)` is unique to builtin components.